eZ/Publish/API/Repository/Tests/Values/User/Limitation/LanguageLimitationTest.php

```diff
@@ -490,6 +493,129 @@ public function testPublishVersionTranslationWhenUserHasNoAccessToAllLanguages()
+
+        self::assertSame($content->getVersionInfo()->languageCodes, $clonedContent->getVersionInfo()->languageCodes);
+    }
+
+    /**
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     * @throws \eZ\Publish\API\Repository\Exceptions\BadStateException
+     * @throws \eZ\Publish\API\Repository\Exceptions\ForbiddenException
+     * @throws \eZ\Publish\API\Repository\Exceptions\InvalidArgumentException
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     */
+    public function testMoveSubtreeWithLanguageLimitationDifferentThanInitialLanguageCode(): void
+    {
+        $repository = $this->getRepository();
+        $locationService = $repository->getLocationService();
+        $contentService = $repository->getContentService();
+
+        $content = $this->prepareDataForLocationRelatedTests();
+
+        $contentLocation = $locationService->loadLocation($content->contentInfo->mainLocationId);
+        $targetLocation = $locationService->loadLocation(52);
+        $locationService->moveSubtree($contentLocation, $targetLocation);
+
+        $movedContent = $contentService->loadContent($content->id);
+
+        self::assertSame($content->getVersionInfo()->languageCodes, $movedContent->getVersionInfo()->languageCodes);
+    }
+
+    /**
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     * @throws \eZ\Publish\API\Repository\Exceptions\BadStateException
+     * @throws \eZ\Publish\API\Repository\Exceptions\ForbiddenException
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     */
+    public function testHideUnhideLocationWithLanguageLimitationDifferentThanInitialLanguageCode(): void
+    {
+        $repository = $this->getRepository();
+        $locationService = $repository->getLocationService();
+
+        $content = $this->prepareDataForLocationRelatedTests();
+
+        $location = $locationService->loadLocation($content->contentInfo->mainLocationId);
+        $locationService->hideLocation($location);
+
+        $hiddenLocation = $locationService->loadLocation($location->id);
+
+        self::assertSame(
+            $content->getVersionInfo()->languageCodes,
+            $hiddenLocation->getContent()->getVersionInfo()->languageCodes
+        );
+
+        $visibleLocation = $locationService->unhideLocation($hiddenLocation);
+
+        self::assertSame(
+            $content->getVersionInfo()->languageCodes,
+            $visibleLocation->getContent()->getVersionInfo()->languageCodes
+        );
+    }
+
+    /**
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     * @throws \eZ\Publish\API\Repository\Exceptions\BadStateException
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     * @throws \eZ\Publish\API\Repository\Exceptions\ForbiddenException
+     */
+    private function prepareDataForLocationRelatedTests(): Content
```

Comment:
Following the previous comment, this should be less generic. E.g:
```suggestion
    private function prepareDataForTestsOfLimitationWithDifferentInitialLanguageCode(): Content
```
The name is a bit too long, but ATM have no better idea how to describe it briefly ;)